### ISSUE TITLE
Make financial supporters visible.

### ISSUE THREAD

**Issue by egypturnash** [tootsuite/mastodon#5380]
A lot of Mastodon instances are now financially supported by Patreon. What if there was a way to link this and have some marker show up automatically?

Perhaps it might be something as simple as the "⭐️ I support Metafilter!" links that appear on the [profile page](https://www.metafilter.com/user/79202) of people who donate to that venerable site.

Perhaps it might change more things - what if a supporter's user name is displayed as "⭐️chickenface" instead of "@chickenface" throughout the system? What if a supporter got some other perk like a 6000-character toot length limit, invitations to local get-togethers, auto-acceptance to the admin's Secret Nudes Account, whatever? (This could be generalized into a concept of different "user levels" with different capabilities, and different symbols ala [DeviantArt in days of yore](https://mushir.deviantart.com/art/deviantART-Symbols-Explained-156394748)... if there's an issue discussing that concept, I couldn't find it.)

I don't have any tech suggestions for how to implement this beyond noting that [Patreon does have an API](https://docs.patreon.com/#introduction)

(There are other ways of letting users help pay to support an instance, and ultimately it would be great to have as many as possible hooked up to this, but I'd like to suggest keeping the initial scope constrained to Patreon.)

**Comment by nolanlawson** [comment]
Would supporting particular proprietary services like Patreon be against our principles of decentralization? I recall we had a similar discussion about using Gravatars.

Otherwise sounds really reasonable and a good way to help support instances; if nothing else individual instances may want to apply it as a patch, in the same way that e.g. cybre.space provides insta-access to supporters.

**Comment by m4sk1n** [comment]
There should be something like mastodon-extras (maybe `mastodon-ugly` ;)) containing implementations of closed API-s for translation services, Patreon etc.
Also, @nolanlawson, what about Libravatar instead of Gravatar?

**Comment by kevquirk** [comment]
+1 for this. Personally, I would love it if there was a way to call out the people who support our instance. Kind of like a Twitter "verified" badge where an admin/mod can enable it within the user's profile.

**Comment by m4sk1n** [comment]
Maybe that’s not a good source of inspiration, but afaik newgab has donor and investor badges :D

**Comment by kevquirk** [comment]
Has there been any movement on this? It's would be a great feature to have within Mastodon.

**Comment by FediVideos** [comment]
I don't think this should be linked to any particular payment platform, but as a general idea having supporters' badges on profiles (or something similar to badges) sounds like a really good idea.

Admins would be able to verify through payment platforms who has donated, people can leave messages with their donation saying their username and whether they would like a badge.

Maybe there could be a variety of badges for people to collect, or something similar?

I realise not everyone wants badges, or to even say that they've donated, but if it was optional this might be a good way of getting more people interested in donating. If people see others round them with badges, they might ask how they can get one too?

**Comment by raphaelsiz** [comment]
I think, as someone who can't afford to donate, it's fine if there are extra features to be unlocked as long as they're not accessibility features or things that are currently part of Mastodon
like, if they set the character limit to 140 and have it be normal for supporters, that's terrible, or if they introduce settings that would be super helpful for accessibility and they put those behind the pledge I'd throw a fit
but small harmless things like extra emojis or a slightly longer character limit would probably be fine
also, most things are customizable from the instance level anyway, and unlike centralized platforms, admins have no motivation to make the user experience miserable for anyone who isn't paying
if an admin manages to make a version of Mastodon where all the good features are behind a paywall (and I guarantee they *could* do that, even if there are no additional features granted on this repo), people who can't or don't want to pay have a bunch of other instances to choose from, so I don't think many admins would choose to do that

**Comment by ruario** [comment]
I strongly support this. I don't want to see another popular instance go down for lack of funding when it actually has lots of users. I have seen requests from popular instances like [mstdn.social](https://mstdn.social/) asking some support and appearing to struggle to get the word out. Not much is really needed but it feels like the message is skipping over people. Somehow this needs to be highlighted better.

**Comment by johannab33** [comment]
Would love to see an easily integrated "click here to donate" component in the web interface, as well as a way to acknowledge supporters. I'm struggling with this now in the instance I took over, we need to raise some small amount of funds but I'm being stymied as to how to make it part of the instance's home page in an obvious way (trying to use Stripe in our case).

**Comment by kevquirk** [comment]
I'm one of the admins of Fosstodon, which is a fairly large (~20k users) instance. I think a lot of the larger instances live or die by the donations from their users. So if a there's a way to celebrate those people who donate, with something simple like a "donator" or "supporter" tag next to their username, would really help. If there's a way to implement a donations from within Masto, that would be a bonus.

**Comment by JustDalek** [comment]
Ah, just the post I seek. I too would be interested in role sync behavior to thank supporters.

I was hoping some sort of API or webhook that allowed us to manage roles. It would be great to also show some love to certain discord role users through the same config
